### ISSUE TITLE
Dynamic quantization of Albert model

### ISSUE THREAD

**Issue by araabovyan** [microsoft/onnxruntime#11701]
**Dynamicly quantized AlBert model poor performance **
When I quantize fine-tuned albert-base-v2 model F1 score drops to 0.05 whereas on non quantized  version it is 0.8.   

**System information**
- OS Platform: Ubuntu (20.04)
- onnx == 1.11.0     
- onnxruntime == 1.11.1 
- Python version: 3.8.10

`quantize_dynamic(onnx_model_path,
                     output_name,
                     weight_type=QuantType.QInt8)
`

I assume that dynamic quantization adds ops to quantize on each layer of encoder part, but because Albert model has shared layers, it quantizes same  weights several times (to be precise 12 times). 

I tried to quantize only unique layers  by providing _nodes_to_quantize_ argument and close to original results but when I tried reduce model size by using solution from here https://github.com/onnx/onnx/issues/3278#issuecomment-781948998 and got same poor results.
Thanks in advance!

**Comment by araabovyan** [comment]
I have fine-tuned on a token classification task, but I believe that If you quantize only albert base without fine-tuning and compare the outputs, they will be very different.

**Comment by kiran-cq** [comment]
Facing the same issue on a different model: https://github.com/open-mmlab/mmocr/tree/main/configs/textrecog/satrn.
This is an OCR model. Had to remove the shared weights as the converted .onnx file was much bigger than the pytorch weights file; ~650 MB vs ~45 MB. On removal of the shared weights using the method mentioned in https://github.com/onnx/onnx/issues/3278#issuecomment-781948998  the word-accuracy decreases from ~90% to ~70%.
